packages/fx-core/src/plugins/solution/fx-solution/solution.ts

```diff
@@ -1641,13 +1641,11 @@ export class TeamsAppSolution implements Solution {
     const notifications: string[] = [];
     const pluginsToScaffold: LoadedPlugin[] = [this.LocalDebugPlugin];
     const azureResource = Array.from(settings.azureResources || []);
-    if (addFunc || ((addSQL || addApim) && !alreadyHaveFunction)) {
+    if (addFunc && !alreadyHaveFunction) {
```

Comment:
we may also need to handle cases on addFunc = true and alreadyHaveFunction = true, which only scaffolded an new api to Azure Functions project

Reply:
sure, restore the logic